### ISSUE TITLE
describedBy : Cannot convert undefined or null to object

### ISSUE THREAD

**Issue by partikule** [json-editor/json-editor#786]
- json-editor version: 2.3.0

The describedBy example schema hangs. (see console)
https://json-editor.github.io/json-editor/describedby.html

I tried with the master branch, it hangs the same way.

**Comment by partikule** [comment]
At the moment, I just unactivated the links validation in validator.js, method `_validateSchema()`
Not a good solution, but it works.

`
    /*
    if (schema.links) {
      schema.links.forEach((s, m) => {
        if (s.rel && s.rel.toLowerCase() === 'describedby') {
          schema = this._expandSchemaLink(schema, m)
          errors.push(...this._validateSchema(schema, value, path, this.translate))
        }
      })
    }
    */

`

**Comment by partikule** [comment]
The problem remains on JSON result build, and my first correction proposal was not good.

Apparently, the problem appears when the schema changes after ***propertyTypeOptions*** switch.
Switching the **describedBy** changes the schema.
But what happens if the user has already filled the JSON?

New solution : 
- Forget about my first proposal
- Add the following check on the ***source*** var in ***utilities.js***

```
export function extend (destination, ...args) {
  args.forEach(source => {
      if (source)
      {
          Object.keys(source).forEach(property => {
              if (source[property] && isPlainObject(source[property])) {
                  if (!hasOwnProperty(destination, property)) destination[property] = {}
                  extend(destination[property], source[property])
              } else if (Array.isArray(source[property])) {
                  destination[property] = deepCopy(source[property])
              } else {
                  destination[property] = source[property]
              }
          })
      }
  })

  return destination
}

```

**Comment by schmunk42** [comment]
fixed by #808
